src/OpenTK.Windowing.Common/Input/JoystickState.cs

```diff
@@ -211,6 +211,40 @@ public override int GetHashCode()
             return hashCode;
         }
 
+        /// <summary>
+        /// Creates a copy with new arrays.
+        /// </summary>
+        /// <returns>A completely new <see cref="JoystickState"/>.</returns>
+        public object Clone()
+        {
+            Hat[] hats = null;
+            float[] axes = null;
+            bool[] buttons = null;
+
+            if (this._hats != null)
+            {
+                hats = new Hat[this._hats.Length];
+
+                Array.Copy(this._hats, hats, hats.Length);
+            }
+
+            if (this._axes != null)
+            {
+                axes = new float[this._axes.Length];
+
+                Array.Copy(this._axes, axes, axes.Length);
+            }
+
+            if (this._buttons != null)
+            {
+                buttons = new bool[this._buttons.Length];
+
+                Array.Copy(this._buttons, buttons, buttons.Length);
+            }
+
+            return new JoystickState(hats, axes, buttons, Id, Name);
+        }
```

Comment:
I don't like how this method allocates new arrays like this. 
I would much rather have a `void CopyTo(ref JoystickState)` and that way we could without allocations transfer the data from one `JoystickState[]` to another.
Could also be `void CopyTo(JoystickState)` without the `ref`, but the `ref` helps with understanding that the `CopyTo` actually modifies the data referenced by the struct (i.e. that the operation has side-effects on the struct you pass even if it's by value).